enso-org/enso
# Move Project Manager into cached React Query to ensure it is globally unique

### Pull Request Description
- This prevents two separate Project Managers from being created, due to `React.useEffect` firing twice when Strict Mode is enabled.
- Fix crashes with `npm run dev` in `app/gui2` when there are broken symlinks in the projects directory

### Important Notes
None

### Testing Instructions
- Inspect the websockets with address `127.0.0.1:30535` to make sure that only one of them has any messages at all
- Create a broken symlink in the projects directory and confirm that the Local Backend does not fail on this PR (but does on develop)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~
